modules/experimental-layers/src/mesh-layer/mesh-layer.js

```diff
@@ -123,6 +128,17 @@ export default class MeshLayer extends Layer {
 
   initializeState() {
     const attributeManager = this.getAttributeManager();
+
+    this.state.matrixData = new Float32Array(this.getNumInstances() * 16);
+    this.state.matrixBuffer = new Buffer(this.context.gl, this.state.matrixData.byteLength);
+
+    this.state.buffers = {
```

Comment:
Alternatively we can use the attributeManager to handle the creation and update of the matrix buffer:
- Add a single attribute `instanceModelMatCol1: {size: 16}` to the attribute manager
- Override the base layer `updateAttributes` method with [this trick](https://github.com/uber/deck.gl/blob/master/modules/layers/src/solid-polygon-layer/solid-polygon-layer.js#L232) to set each column.

Reply:
> * Override the base layer `updateAttributes` method with [this trick](https://github.com/uber/deck.gl/blob/master/modules/layers/src/solid-polygon-layer/solid-polygon-layer.js#L232) to set each column.

Yes we need to do something like that... though I would prefer if we could store the resulting attributes in the attribute manager rather than directly change the model. I quickly get confused when almost similar things are done/stored in multiple places.

Maybe a cheap initial way would be to can add an `attributeManager.setAttributes()`, do the same calculations but at least store them in the attribute manager?

Especially as this is now happening in multiple layers and is no longer a "one off"...